### ISSUE TITLE
maven-javadoc-plugin fails because eo-runtime.jar is absent in the list of dependencies

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo#1547]
For example, here: https://github.com/objectionary/eo-math/issues/99#issuecomment-1342832357

The problem has been reported earlier, but there was no answer: https://stackoverflow.com/questions/69770376/maven-javadoc-plugin-package-doesnt-exist-when-an-external-jar-is-used

**Comment by yegor256** [comment]
@volodya-lombrozo in order to reproduce the error, try to run `mvn javadoc:jar -Psonatype` in `eo-math` repository, for example

**Comment by volodya-lombrozo** [comment]
`eo-runtime.jar` (version `0.28.11`) doesn't contain any Java classes. We have to find the root problem why it is so. `eo-runtime.jar` with version `0.28.10` contains all required classes.

**Comment by volodya-lombrozo** [comment]
Deployment log [here](https://www.rultor.com/t/28393-1310472258).

The important entries:
```
[1;34mINFOm] 1m--- 0;32meo-maven-plugin:0.28.11:copym 1m(compile)m @ 36meo-runtime0;1m ---m
[1;34mINFOm] 20 source(s) copied from ./home/r/repo/eo-runtime/src/main/eo to ./home/r/repo/eo-runtime/target/classes/EO-SOURCES
[1;34mINFOm]
[1;34mINFOm] 1m--- 0;32meo-maven-plugin:0.28.11:unplacem 1m(compile)m @ 36meo-runtime0;1m ---m
[1;33mWARNINGm] The binary org/eolang/version.txt of org.eolang/eo-runtime/-/0.28.10 looks different, won't unplace
[1;33mWARNINGm] The binary org/eolang/org-eolang-eo-runtime-jar.txt of org.eolang/eo-runtime/-/0.28.10 looks different, won't unplace
[1;33mWARNINGm] The binary EOorg/org-eolang-eo-runtime-jar.txt of org.eolang/eo-runtime/-/0.28.10 looks different, won't unplace
[1;33mWARNINGm] The binary EOorg/EOeolang/EOio/org-eolang-eo-runtime-jar.txt of org.eolang/eo-runtime/-/0.28.10 looks different, won't unplace
[1;33mWARNINGm] The binary EOorg/EOeolang/org-eolang-eo-runtime-jar.txt of org.eolang/eo-runtime/-/0.28.10 looks different, won't unplace
[1;34mINFOm] Just 158 binari(es) out of 163 deleted in ./home/r/repo/eo-runtime/target/eo/placed.csv
[1;34mINFOm]
[1;34mINFOm] 1m--- 0;32meo-maven-plugin:0.28.11:unspilem 1m(compile)m @ 36meo-runtime0;1m ---m
[1;34mINFOm] Deleted 183 .class files out of 183 in ./home/r/repo/eo-runtime/target/classes
[1;34mINFOm]
[1;34mINFOm] 1m--- 0;32mmaven-jar-plugin:3.3.0:jarm 1m(default-jar)m @ 36meo-runtime0;1m ---m
[1;34mINFOm] Building jar: /home/r/repo/eo-runtime/target/eo-runtime-0.28.11.jar
```
Probably `eo-maven-plugin:0.28.11:unspile` removed all files from `target` folder before packaging.

**Comment by volodya-lombrozo** [comment]
@yegor256 I can't reproduce the problem locally, but in logs I've seen the next command:
```sh
mvn clean deploy -Pobjectionary -Psonatype --errors --settings ../settings.xml -Dstyle.color=never
```
Can you share the `settings.xml` content please?

**Comment by volodya-lombrozo** [comment]
Finally, I’ve investigated the root of the problem. The problem started happening after implementing that [issue](https://github.com/objectionary/eo/issues/1361). More precisely, after that [PR](https://github.com/objectionary/eo/pull/1357).
The problem is the next:
1.  We started to add `eo-runtime` library for all eo programs on `ResolveMojo` since all classes from `eo-runtime` are required for further compilation almost all eo programs (see the issue for more details).
2.  Since all `eo-runtime` classes were added to `target` directory, `PlaceMojo` stage looks on them and add to `placedTojos` list.    
3.  `UnplaceMojo` removes all tojos that were placed on `PlaceMojo` stage when we build a new version of `eo-runtime` library itself => removes all new `eo-runtime` compiled classes

In other words, compilation works for all eo programs, instead of `eo-runtime` library itself. I’ll think about the proper fix.

**Comment by yegor256** [comment]
@volodya-lombrozo maybe it was a bad idea to add eo-runtime to the list of dependencies quietly, behind the scene? If we roll back this functionality, everything will work just fine. I'm not a big fan of anything to happen "by default". It's better to make the user be explicit as much as possible.

**Comment by volodya-lombrozo** [comment]
> @volodya-lombrozo maybe it was a bad idea to add eo-runtime to the list of dependencies quietly, behind the scene? If we roll back this functionality, everything will work just fine. I'm not a big fan of anything to happen "by default". It's better to make the user be explicit as much as possible.

Maybe you right. The problem here that we need it sometimes. Absence of the library leads to some bugs. Actuality I have the PR #1576 that solves the issue. The only problem with the fix - I don't know how to prove it automatically yet #1575.

**Comment by yegor256** [comment]
@volodya-lombrozo we can release a new version of this product, what do you think?

**Comment by volodya-lombrozo** [comment]
@yegor256 I think we have to do so. At least in order to fix the current issue.

**Comment by yegor256** [comment]
@rultor release, tag is `0.28.12`

**Comment by yegor256** [comment]
@rultor release, tag is `0.28.12`
